src/Meltor.php

```diff
@@ -0,0 +1,872 @@
+    {
+        $query = 'SELECT
+                    stat.TABLE_SCHEMA AS database_name, 
+                    stat.TABLE_NAME, 
+                    stat.INDEX_NAME, 
+                    GROUP_CONCAT(stat.COLUMN_NAME ORDER BY stat.seq_in_index separator ", ") AS columns, 
+	                GROUP_CONCAT(IFNULL(stat.SUB_PART, "NULL") ORDER BY stat.seq_in_index separator ", ") AS lengths, 
+                    constraints.CONSTRAINT_TYPE
+                FROM information_schema.STATISTICS stat
+                     JOIN information_schema.table_constraints constraints
+                      ON stat.TABLE_SCHEMA = constraints.TABLE_SCHEMA
+                          AND stat.TABLE_NAME = constraints.TABLE_NAME
+                          AND stat.INDEX_NAME = constraints.CONSTRAINT_NAME
+
+                WHERE stat.NON_UNIQUE = 0
+                  AND stat.TABLE_SCHEMA = "%s"
+                  AND constraints.CONSTRAINT_TYPE = "UNIQUE"
+
+                GROUP BY stat.TABLE_SCHEMA,
+                         stat.TABLE_NAME,
+                         stat.INDEX_NAME,
+                         constraints.CONSTRAINT_TYPE
+
+                ORDER BY stat.TABLE_SCHEMA,
+                         stat.TABLE_NAME;';
+
+        return collect($connection->select(sprintf($query, $databaseName)))->groupBy('TABLE_NAME');
+    }
+
+    /**
+     * Returns a list of foreign keys for the given MySQL database.
+     *
+     * @param string $databaseName
+     * @param ConnectionInterface $connection
+     * @return Collection
+     */
+    public function getForeignKeys(string $databaseName, ConnectionInterface $connection): Collection
+    {
+        $query = 'SELECT 
+                    `usages`.`TABLE_NAME`,
+                    `usages`.`COLUMN_NAME`,
+                    `usages`.`CONSTRAINT_NAME`,
+                    `usages`.`REFERENCED_TABLE_NAME`, 
+                    `usages`.`REFERENCED_COLUMN_NAME`, 
+                    `foreigns`.`TYPE`
+                 FROM `INFORMATION_SCHEMA`.`KEY_COLUMN_USAGE` AS `usages` 
+                 INNER JOIN `INNODB_FOREIGN` AS `foreigns` ON `foreigns`.`ID` = CONCAT(`usages`.`CONSTRAINT_SCHEMA`, "/", `usages`.`CONSTRAINT_NAME`) 
+                 WHERE `usages`.`REFERENCED_TABLE_SCHEMA` = "%s";';
+
+        return collect($connection->select(sprintf($query, $databaseName)))->groupBy('TABLE_NAME');
+    }
+
+    /**
+     * Extract the MySQL DATA_TYPE column property and check it.
+     *
+     * @param object $column
+     * @param bool $ignoreProblems
+     * @return string|null
+     * @throws Exception
+     */
+    protected function getDataType(object $column, bool $ignoreProblems = false): ?string
+    {
+        $columnType = $column->COLUMN_TYPE;
+        $dataType   = $column->DATA_TYPE;
+
+        if ($columnType == 'tinyint(1)') {
+            return 'boolean';
+        }
+
+        if (!array_key_exists($dataType, $this->config('mysql.fluentDataTypes'))) {
+            if (!$ignoreProblems) {
```

Comment:
To be honest, I would rather just throw exceptions everywhere and let the caller catch them instead of doing those ignoreProblems checks all over the place